metagpt/roles/ui_designer.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+# @Date    : 2023/8/16 13:58
+# @Author  : stellahong (stellahong@fuzhi.ai)
+# @Desc    :
+from functools import wraps
+import json5
+
+from metagpt.logs import logger
+from metagpt.roles import Role
+from metagpt.schema import Message
+
+from metagpt.actions.design import Tool, SDPromptExtend, SDPromptOptimize, SDPromptImprove
+from metagpt.actions.ui_design import ModelSelection, SDGeneration
+
+
+def retrieve(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        content, keyword = func(*args, **kwargs)
+        info = content.replace(keyword, "")
+        return info
+    
+    return wrapper
+
+
+class Designer(Role):
+    """Class representing the UI designer Role."""
+    
+    def __init__(
+            self,
+            name="Catherine",
+            profile="UI Design",
+            goal="Generate UI icon",
+            constraints="Give clear icon description and generate images to finish the design",
+            actions=[ModelSelection, SDPromptExtend, SDGeneration]):
+        super().__init__(name, profile, goal, constraints)
+        
+        self._init_actions(actions)
+
+    @property
+    def memory_model_name(self):
+        return "MODEL_NAME: "
+    
+    @property
+    def memory_user_input(self):
+        return "User Input: "
+    
+    @property
+    def memory_domain(self):
+        return "Domain: "
+    
+    def memory_property(self, memory_keyword: str, memory_content: str):
+        self._rc.memory.add(Message(f"{memory_keyword}{memory_content}", role=self.profile))
+    
+    @retrieve
+    def get_important_memory(self, keyword: str):
+        query_memory = self._rc.memory.get_by_content(keyword)[0]
+        return query_memory.content, keyword
+    
+    async def _plan_and_select(self):
+        """
+        这里实现的是二选一的option，action在这里进行了选择
+        理论上应该可以实现4种选择 (&：表示串行顺序），目前只选择了前2种
+        1）action1
```

Comment:
Inside the comment, the ")" and ")" symbols are mixed

Reply:
ok